### PR TITLE
Json net 7

### PR DESCRIPTION
**Pull request by rogeralsing**
This PR introduces Json.NET and removes the old DU hacks we had in the serializer.

This closes: #1127 #1126 #999 #827

### CODE REVIEW

File: src/core/Akka/Serialization/NewtonSoftJsonSerializer.cs [hunk 1]
```diff
@@ -107,40 +110,32 @@ public override byte[] ToBinary(object obj)
...
-                    var paramTypes = ctor.GetParameters().Select(p => p.ParameterType).ToArray();
-                    var values =
-                        j["Fields"].ToObject<object[]>()
-                            .Select((o, i) => TranslateSurrogate(o, system, paramTypes[i]))
-                            .ToArray();
-                    var res = ctor.Invoke(values.ToArray());
-                    return res;
-                }
+                //The JObject is not of our concern, let Json.NET deserialize it.
+                return j.ToObject(type, parent._serializer);
```
Comment (rogeralsing) [inline src/core/Akka/Serialization/NewtonSoftJsonSerializer.cs]: This does the magic, if we get a JObject, we can pass in the same settings as we use for other serialzation, and things turns out correct :)

**Comment by Aaronontheweb** [comment]
I'm kind of the opinion that we should just pull this in anyway buuuuuuuuuut 

image

What sort of risk is there for this change to cause Akka.NET to be incompatible with other stuff that depends on JSON.NET, such as ASP.NET MVC?

**Comment by rogeralsing** [comment]
@Aaronontheweb I have absolutely no idea :) maybe a simple assembly version redirect would work in those cases. ¯\_(ツ)_/¯

**Comment by Aaronontheweb** [comment]
A BRAVE NEW WORLD

**Pull request merged**